### PR TITLE
♻️ Remove use of `.find`

### PR DESCRIPTION
**Pull request by alanorozco**
- Fixes compatibility in very old browsers that don't support `.find()`
- Fixes https://github.com/ampproject/error-reporting/issues/103

### CODE REVIEW

File: extensions/amp-date-display/format.js [hunk 1]
```diff
@@ -19,8 +19,11 @@ export function getTimeZoneName(date, locale, options, format = 'long') {
...
     timeZoneName: format,
   });
-  return (
-    formatter.formatToParts(date).find(({type}) => type === 'timeZoneName')
-      ?.value || ''
-  );
+  const parts = formatter.formatToParts?.(date) || [];
+  for (let i = 0; i < parts.length; i++) {
+    if (parts[i].type === 'timeZoneName') {
+      return parts[i].value;
```
Comment (jridgewell) [inline extensions/amp-date-display/format.js:25]: Nit:

```suggestion
  for (const part of parts) {
    if (part.type === 'timeZoneName') {
      return part.value;
```

**Review by jridgewell — APPROVED** [review]

**Comment by samouri** [comment]
Should we add an eslint rule against .find()?

**Comment by alanorozco** [comment]
@samouri We cannot do it with a lint rule since it requires knowing that the object is an `Array` and not a different type. Suggestions welcome.

**Comment by samouri** [comment]
@alanorozco, good point. a simple eslint rule would end up being naive and flag much too much. in order to properly translate the conformance checks away from closure would require using the TS API to verify types

**Pull request merged**